### PR TITLE
Saumya branch

### PR DESCRIPTION
**Pull request by Saumya-Shah**
MPC evaluation, with gym, updated for the new forward model

### CODE REVIEW

File: mpc_eval.py [hunk 1]
```diff
@@ -124,51 +121,50 @@ def diverse_sampling(code):
...
-            state_cur_mpc = fwd_model_decoder(state_action_concate, feats_fwd)
             best_action_list.append(best_action_so_far)
 
             image_error = mse(state_cur_mpc, state_fut)
 
             # Cumulative action error with diverse samples
             image_error_sum += image_error
             step += 1
-        action_hat = torch.cat(best_action_list,dim=1)
+        action_hat = torch.cat(best_action_list,dim=0)
```
Comment (goodmattg) [inline mpc_eval.py:167]: Why did this change?

File: mpc_eval.py [hunk 2]
```diff
@@ -102,7 +99,7 @@ def diverse_sampling(code):
             actions.float().to(gpu_id),
             goal.float().to(gpu_id),
         )
-        print(dataset.seq_length)
+        print("trajectory: ", i)
```
Comment (goodmattg) [inline mpc_eval.py:102]: Do you mind using _logging_ instead of print statements?

File: generate_trajectories.py [hunk 3]
```diff
@@ -73,6 +88,8 @@ def generate_trajectory(env, actor_network, args):
             env.env.goal = np.hstack(
                 [env.initial_gripper_xpos[:2].copy() + goal_offset, env.height_offset]
             )
+        
+        print("hiiiii",env.env.goal.shape)
```
Comment (goodmattg) [inline generate_trajectories.py:92]: remove this

File: generate_trajectories.py [hunk 4]
```diff
@@ -109,7 +126,7 @@ def generate_trajectory(env, actor_network, args):
         with torch.no_grad():
             pi = actor_network(inputs)
         action = pi.detach().numpy().squeeze()
-
+        print(action.shape)
```
Comment (goodmattg) [inline generate_trajectories.py:129]: remove this

File: control_evaluation.py [hunk 5]
```diff
@@ -130,19 +127,15 @@ def diverse_sampling(code):
...
+
+            state_fut_hat = fwd_model_autoencoder(state_cur_fwd,action_hat[:, image_num])
             state_cur_fwd = state_fut_hat
 
             image_error = mse(state_fut_hat, state_fut)
             # Cumulative action error with diverse samples
             image_error_sum += image_error
             step += 1
-
+            print(image_num)
```
Comment (goodmattg) [inline control_evaluation.py:138]: remove this

File: MPC_gym_eval.py [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+        )
+        
+        
+        actions = actions[:,:-1,:]
+
+        # print(state_cur_fwd.size())
+        image_error_sum = 0
+        best_action_list = []
+        for image_num in range(dataset.seq_length - 1):
+            print(image_num)
```
Comment (goodmattg) [inline MPC_gym_eval.py:173]: please remove

File: mpc_eval.py (hunk 1 above)
Comment (Saumya-Shah) [reply mpc_eval.py:167]: It isn't used as such, was just for MSE error calculation. Had a shape error so I had to flatten it instead of stack.

**Pull request merged**